Fixing device tree for irq chip initialization for mz7020

wpatterson-witekio opened the pull request.

Commit:
Fixing device tree for irq chip initialization for mz7020

narrucmot reviewed (commented):
I don't believe this commit is necessary.  This devicetree edit correlates to a couple of settings for the axi_intc_0 IP in the Vivado hw project that 1) set the interrupt type to "edge" and 2) set the interrupt output connection to "single".  From the board Vivado TCL script:
   \# Set IRQ type to 'EDGE' and connection to 'SINGLE'
   set_property -dict [ list \
      CONFIG.C_IRQ_IS_LEVEL {0} \
      CONFIG.C_IRQ_CONNECTION {1}] [get_bd_cells axi_intc_0]

![image](https://user-images.githubusercontent.com/12900664/145485589-de8f53b2-c011-4143-8881-3e686acb3447.png)

With these settings in the Vivado script - required for Vitis and Vitis-AI - the Linux OS boots as expected and no changes needed to the devicetree:
/* Vitis interrupt connected at PS interrupt offset [0] (29+0=29) */
/* This description may be missing/incorrect in the pl.dtsi, so we add/fix it here */
&axi_intc_0 {
   interrupts = <0 29 1>; 
};

--Tom

wpatterson-witekio commented:
I don't quite follow. 

This line that I added is a typo, which I've removed and verified locally
`   interrupts = <0 29 4>;`
interrupts is already defined, so I don't want to redefine it again


In final testing pz, I found this same issue with all pz boards, which was not there in the initial testing, so something changed.
The error is:
```
smp: Bringing up secondary CPUs ...
8<--- cut here ---
Unable to handle kernel NULL pointer dereference at virtual address 00000008
pgd = (ptrval)
[00000008] *pgd=00000000
Internal error: Oops - BUG: 805 [#1] PREEMPT SMP ARM
Modules linked in:
CPU: 1 PID: 0 Comm: swapper/1 Not tainted 5.10.0-xilinx-v2021.1 #1
Hardware name: Xilinx Zynq Platform
PC is at xintc_write+0x58/0x80
LR is at arm_heavy_mb+0x18/0x4c
pc : [<c037e6b0>]    lr : [<c0111d78>]    psr: 600001d3
sp : c0c8bf98  ip : 00000020  fp : 00000000
r10: 00000000  r9 : 413fc090  r8 : 0000406a
r7 : 00000000  r6 : 00000008  r5 : 00000000  r4 : eeef2140
r3 : 00000000  r2 : 00000000  r1 : 00000000  r0 : f0808730
Flags: nZCv  IRQs off  FIQs off  Mode SVC_32  ISA ARM  Segment none
Control: 18c5387d  Table: 0000404a  DAC: 00000051
Process swapper/1 (pid: 0, stack limit = 0x(ptrval))
Stack: (0xc0c8bf98 to 0xc0c8c000)
bf80:                                                       eeef2140 00000001
bfa0: 00000091 c037e8dc eeeec29c 00000001 00000091 00000000 0000406a c037e99c
bfc0: eeeec29c c011cef8 00000000 00000047 00000001 c0a1f738 c0c8a000 00000001
bfe0: c0a42350 c010b7b0 00c8006a 00000051 10c0387d 001014ac 00000000 00000000
[<c037e6b0>] (xintc_write) from [<c037e8dc>] (xil_intc_initial_setup+0x14/0xbc)
[<c037e8dc>] (xil_intc_initial_setup) from [<c037e99c>] (xil_intc_start+0x18/0x28)
[<c037e99c>] (xil_intc_start) from [<c011cef8>] (notify_cpu_starting+0x6c/0x7c)
[<c011cef8>] (notify_cpu_starting) from [<c010b7b0>] (secondary_start_kernel+0x104/0x170)
[<c010b7b0>] (secondary_start_kernel) from [<001014ac>] (0x1014ac)
Code: ebf64dae e5941000 e6bf5f35 e0816006 (e5865000) 
---[ end trace 59a0fb72d008c3e5 ]---
Kernel panic - not syncing: Attempted to kill the idle task!
---[ end Kernel panic - not syncing: Attempted to kill the idle task! ]---
CPU0: stopping

```

I initially found this testing mz7010 and mz7020, where mz7010 booted successfully and mz7020 failed. By comparing the logs I was able to trace the failure to 2 different initialization paths in the driver which diverged depending on weather or not there was an interrupt-parent. I then decompiled both device tree blobs and did a diff and saw the only difference was the failing device tree was missing the interrupt-parent and interrupt name lines. Adding these 2 lines fixes the issue for mz7020 and all pz boards.

narrucmot commented:
This change to the Vivado board TCL script is recent and was part of the PR that I merged a few days ago:

   # Set IRQ type to 'EDGE' and connection to 'SINGLE'
   set_property -dict [ list \
      CONFIG.C_IRQ_IS_LEVEL {0} \
      CONFIG.C_IRQ_CONNECTION {1}] [get_bd_cells axi_intc_0]

If you did any Vivado builds and testing before that, then you were likely to see the boot failure you described, and then fixed with the “interrupts = <0 29 4>” line, since that would have matched the axi_intc_0 Vivado settings at that time..  The ‘interrupt-names’ and ‘interrupts’ lines aren’t needed in the system-bsp.dtsi file since they are previously defined in the auto-generated pl.dtsi in the PetaLinux project:
       amba_pl: amba_pl {
              #address-cells = <1>;
              #size-cells = <1>;
              compatible = "simple-bus";
              ranges ;
              axi_intc_0: ***@***.*** {
                     #interrupt-cells = <2>;
                     clock-names = "s_axi_aclk";
                     clocks = <&misc_clk_0>;
                     compatible = "xlnx,axi-intc-4.1", "xlnx,xps-intc-1.00.a";
                     interrupt-controller ;
                     interrupt-names = "irq";
                     interrupt-parent = <&intc>;
                     interrupts = <0 29 1>;
                     reg = <0x41800000 0x10000>;
                     xlnx,kind-of-intr = <0x1>;
                     xlnx,num-intr-inputs = <0x20>;

--Tom


From: wpatterson-witekio ***@***.***>
Sent: Thursday, December 9, 2021 5:46 PM
To: Avnet/meta-avnet ***@***.***>
Cc: Curran, Tom (EM) ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [Avnet/meta-avnet] Fixing device tree for irq chip initialization for mz7020 (PR #47)


I don't quite follow.

This line that I added is a typo, which I've removed and verified locally
interrupts = <0 29 4>;

In final testing pz, I found this same issue with all pz boards, which was not there in the initial testing, so something changed.
The error is:

smp: Bringing up secondary CPUs ...

8<--- cut here ---

Unable to handle kernel NULL pointer dereference at virtual address 00000008

pgd = (ptrval)

[00000008] *pgd=00000000

Internal error: Oops - BUG: 805 [#1] PREEMPT SMP ARM

Modules linked in:

CPU: 1 PID: 0 Comm: swapper/1 Not tainted 5.10.0-xilinx-v2021.1 #1

Hardware name: Xilinx Zynq Platform

PC is at xintc_write+0x58/0x80

LR is at arm_heavy_mb+0x18/0x4c

pc : [<c037e6b0>]    lr : [<c0111d78>]    psr: 600001d3

sp : c0c8bf98  ip : 00000020  fp : 00000000

r10: 00000000  r9 : 413fc090  r8 : 0000406a

r7 : 00000000  r6 : 00000008  r5 : 00000000  r4 : eeef2140

r3 : 00000000  r2 : 00000000  r1 : 00000000  r0 : f0808730

Flags: nZCv  IRQs off  FIQs off  Mode SVC_32  ISA ARM  Segment none

Control: 18c5387d  Table: 0000404a  DAC: 00000051

Process swapper/1 (pid: 0, stack limit = 0x(ptrval))

Stack: (0xc0c8bf98 to 0xc0c8c000)

bf80:                                                       eeef2140 00000001

bfa0: 00000091 c037e8dc eeeec29c 00000001 00000091 00000000 0000406a c037e99c

bfc0: eeeec29c c011cef8 00000000 00000047 00000001 c0a1f738 c0c8a000 00000001

bfe0: c0a42350 c010b7b0 00c8006a 00000051 10c0387d 001014ac 00000000 00000000

[<c037e6b0>] (xintc_write) from [<c037e8dc>] (xil_intc_initial_setup+0x14/0xbc)

[<c037e8dc>] (xil_intc_initial_setup) from [<c037e99c>] (xil_intc_start+0x18/0x28)

[<c037e99c>] (xil_intc_start) from [<c011cef8>] (notify_cpu_starting+0x6c/0x7c)

[<c011cef8>] (notify_cpu_starting) from [<c010b7b0>] (secondary_start_kernel+0x104/0x170)

[<c010b7b0>] (secondary_start_kernel) from [<001014ac>] (0x1014ac)

Code: ebf64dae e5941000 e6bf5f35 e0816006 (e5865000)

---[ end trace 59a0fb72d008c3e5 ]---

Kernel panic - not syncing: Attempted to kill the idle task!

---[ end Kernel panic - not syncing: Attempted to kill the idle task! ]---

CPU0: stopping



I initially found this testing mz7010 and mz7020, where mz7010 booted successfully and mz7020 failed. By comparing the logs I was able to trace the failure to 2 different initialization paths in the driver which diverged depending on weather or not there was an interrupt-parent. I then decompiled both device tree blobs and did a diff and saw the only difference was the failing device tree was missing the interrupt-parent and interrupt name lines. Adding these 2 lines fixes the issue for mz7020 and all pz boards.

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/Avnet/meta-avnet/pull/47#issuecomment-990369578>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCNSOECRXKNJPUCJNG7IK3UQEWTTANCNFSM5JULFRCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

wpatterson-witekio commented:
I just rebuilt mz7020 and the auto generated pl.dtsi does not include those lines. I get 
```
/ {
	amba_pl: amba_pl {
		#address-cells = <1>;
		#size-cells = <1>;
		compatible = "simple-bus";
		ranges ;
		axi_intc_0: interrupt-controller@41800000 {
			#interrupt-cells = <2>;
			clock-names = "s_axi_aclk";
			clocks = <&misc_clk_0>;
			compatible = "xlnx,axi-intc-4.1", "xlnx,xps-intc-1.00.a";
			interrupt-controller ;
			reg = <0x41800000 0x10000>;
			xlnx,kind-of-intr = <0x1>;
			xlnx,num-intr-inputs = <0x20>;
		};
```

narrucmot commented:
That is definitely interesting.  I built the MZ7010 last night and the generated pl.dtsi in my PetaLinux project does include those lines (entire amba_pl node shown for completeness):

/ {
       amba_pl: amba_pl {
              #address-cells = <1>;
              #size-cells = <1>;
              compatible = "simple-bus";
              ranges ;
              axi_intc_0: ***@***.*** {
                     #interrupt-cells = <2>;
                     clock-names = "s_axi_aclk";
                     clocks = <&misc_clk_0>;
                     compatible = "xlnx,axi-intc-4.1", "xlnx,xps-intc-1.00.a";
                     interrupt-controller ;
                     interrupt-names = "irq";
                     interrupt-parent = <&intc>;
                     interrupts = <0 29 1>;
                     reg = <0x41800000 0x10000>;
                     xlnx,kind-of-intr = <0x1>;
                     xlnx,num-intr-inputs = <0x20>;
              };
              misc_clk_0: misc_clk_0 {
                     #clock-cells = <0>;
                     clock-frequency = <100000000>;
                     compatible = "fixed-clock";
              };
              zyxclmm_drm {
                     compatible = "xlnx,zocl";
                     interrupts-extended = <&axi_intc_0 0 4>, <&axi_intc_0 1 4>, <&axi_intc_0 2 4>, <&axi_intc_0 3 4>, <&axi_intc_0 4 4>, <&axi_intc_0 5 4>, <&axi_intc_0 6 4>, <&axi_intc_0 7 4>, <&axi_intc_0 8 4>, <&axi_intc_0 9 4>,
<&axi_intc_0 10 4>, <&axi_intc_0 11 4>, <&axi_intc_0 12 4>, <&axi_intc_0 13 4>, <&axi_intc_0 14 4>,
<&axi_intc_0 15 4>, <&axi_intc_0 16 4>, <&axi_intc_0 17 4>, <&axi_intc_0 18 4>, <&axi_intc_0 19 4>,
<&axi_intc_0 20 4>, <&axi_intc_0 21 4>, <&axi_intc_0 22 4>, <&axi_intc_0 23 4>, <&axi_intc_0 24 4>,
<&axi_intc_0 25 4>, <&axi_intc_0 26 4>, <&axi_intc_0 27 4>, <&axi_intc_0 28 4>, <&axi_intc_0 29 4>,
<&axi_intc_0 30 4>, <&axi_intc_0 31 4 >;
              };
       };
};

Why would these lines be in my project’s pl.dtsi file, but not yours?

--Tom


From: wpatterson-witekio ***@***.***>
Sent: Friday, December 10, 2021 2:35 PM
To: Avnet/meta-avnet ***@***.***>
Cc: Curran, Tom (EM) ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [Avnet/meta-avnet] Fixing device tree for irq chip initialization for mz7020 (PR #47)


I just rebuilt mz7010 and the auto generated pl.dtsi does not include those lines. I get

/ {

        amba_pl: amba_pl {

               #address-cells = <1>;

               #size-cells = <1>;

               compatible = "simple-bus";

               ranges ;

               axi_intc_0: ***@***.*** {

                       #interrupt-cells = <2>;

                       clock-names = "s_axi_aclk";

                       clocks = <&misc_clk_0>;

                       compatible = "xlnx,axi-intc-4.1", "xlnx,xps-intc-1.00.a";

                       interrupt-controller ;

                       reg = <0x41800000 0x10000>;

                       xlnx,kind-of-intr = <0x1>;

                       xlnx,num-intr-inputs = <0x20>;

               };

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/Avnet/meta-avnet/pull/47#issuecomment-991241786>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCNSOGLUK3C6HDN735KKZDUQJI7VANCNFSM5JULFRCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

wpatterson-witekio commented:
Try MZ7020 or PZ70XX. The device tree gets built correctly for MZ7010. It's the other boards that are missing those lines.

narrucmot commented:
Hmmm…My MZ7020 project is building now, but I can see already that the devicetree files have been generated and the pl.dtsi file does include the lines in question:

/ {
       amba_pl: amba_pl {
              #address-cells = <1>;
              #size-cells = <1>;
              compatible = "simple-bus";
              ranges ;
              axi_intc_0: ***@***.*** {
                     #interrupt-cells = <2>;
                     clock-names = "s_axi_aclk";
                     clocks = <&misc_clk_0>;
                     compatible = "xlnx,axi-intc-4.1", "xlnx,xps-intc-1.00.a";
                     interrupt-controller ;
                     interrupt-names = "irq";
                     interrupt-parent = <&intc>;
                     interrupts = <0 29 1>;
                     reg = <0x41800000 0x10000>;
                     xlnx,kind-of-intr = <0x1>;
                     xlnx,num-intr-inputs = <0x20>;
              };
              misc_clk_0: misc_clk_0 {
                     #clock-cells = <0>;
                     clock-frequency = <100000000>;
                     compatible = "fixed-clock";
              };
              zyxclmm_drm {
                     compatible = "xlnx,zocl";
                     interrupts-extended = <&axi_intc_0 0 4>, <&axi_intc_0 1 4>, <&axi_intc_0 2 4>, <&axi_intc_0 3 4>, <&axi_intc_0 4 4>, <&axi_intc_0 5 4>, <&axi_intc_0 6 4>, <&axi_intc_0 7 4>, <&axi_intc_0 8 4>, <&axi_intc_0 9 4>,
<&axi_intc_0 10 4>, <&axi_intc_0 11 4>, <&axi_intc_0 12 4>, <&axi_intc_0 13 4>, <&axi_intc_0 14 4>,
<&axi_intc_0 15 4>, <&axi_intc_0 16 4>, <&axi_intc_0 17 4>, <&axi_intc_0 18 4>, <&axi_intc_0 19 4>,
<&axi_intc_0 20 4>, <&axi_intc_0 21 4>, <&axi_intc_0 22 4>, <&axi_intc_0 23 4>, <&axi_intc_0 24 4>,
<&axi_intc_0 25 4>, <&axi_intc_0 26 4>, <&axi_intc_0 27 4>, <&axi_intc_0 28 4>, <&axi_intc_0 29 4>,
<&axi_intc_0 30 4>, <&axi_intc_0 31 4 >;
              };
       };
};

We should figure out why this is not the case for your BSP builds.

--Tom


From: wpatterson-witekio ***@***.***>
Sent: Friday, December 10, 2021 3:19 PM
To: Avnet/meta-avnet ***@***.***>
Cc: Curran, Tom (EM) ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [Avnet/meta-avnet] Fixing device tree for irq chip initialization for mz7020 (PR #47)


Try MZ7020 or PZ70XX. The device tree gets built correctly for MZ7010. It's the other boards that are missing those lines.

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/Avnet/meta-avnet/pull/47#issuecomment-991267671>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCNSOBYFGEICFTIO7JRRRLUQJOERANCNFSM5JULFRCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

wpatterson-witekio commented:
I'm building via `make_mz7020_som_base.sh` Are the timestamps old? Wondering if maybe you have a cached version

narrucmot commented:
I am on the 2021.1 branch of the petalinux and hdl repositories and did a fresh pull this morning.  I don’t have any unstaged or uncommitted local edits.  My common.sh file is set to clone meta-avnet on the 2021.1 branch:

META_AVNET_URL=https://github.com/Avnet/meta-avnet.git
META_AVNET_BRANCH="2021.1"

My Xilinx env vars are all set for the 2021.1 tools:
$ printenv |grep XILINX; printenv |grep PETALINUX; printenv |grep VITIS
XILINX_VITIS=/tools/Xilinx/Vitis/2021.1
XILINX_VIVADO=/tools/Xilinx/Vivado/2021.1
XILINX_HLS=/tools/Xilinx/Vitis_HLS/2021.1
PETALINUX_VER=2021.1
PETALINUX=/tools/Xilinx/PetaLinux/2021.1
XILINX_VITIS=/tools/Xilinx/Vitis/2021.1

--Tom


From: wpatterson-witekio ***@***.***>
Sent: Friday, December 10, 2021 3:52 PM
To: Avnet/meta-avnet ***@***.***>
Cc: Curran, Tom (EM) ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [Avnet/meta-avnet] Fixing device tree for irq chip initialization for mz7020 (PR #47)


I'm building via make_mz7020_som_base.sh Are the timestamps old? Wondering if maybe you have a cached version

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/Avnet/meta-avnet/pull/47#issuecomment-991285216>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCNSOHZ2XMS7Y5TR6BDWWTUQJR5RANCNFSM5JULFRCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

wpatterson-witekio commented:
My setup is the same. What command are you using to build?
If you're not, can you try `make_mz7020_som_base.sh`

As I mentioned, running `make_mz7020_som_base.sh` with all the same setup as you reproduces the issue of those missing lines.
Here's where it gets interesting though. I then checked out my branch `wp-mz7020-fix-irq` in `project-spec/meta-avnet` and ran `rebuild_mz7020_som_base.sh` to see if my changes would be picked up in `pl.dtsi`. Although the timestamp was updated, the changes were not reflected.

This tells me that `pl.dtsi` is not cleanly being regenerated.

narrucmot commented:
Did you run a 'petalinux-build -x mrproper -f' before running the rebuild script?  That needs to be done to clean and restore the project.

--Tom
________________________________
From: wpatterson-witekio ***@***.***>
Sent: Friday, December 10, 2021 4:44:29 PM
To: Avnet/meta-avnet ***@***.***>
Cc: Tom Curran ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [Avnet/meta-avnet] Fixing device tree for irq chip initialization for mz7020 (PR #47)


My setup is the same. What command are you using to build?
If you're not, can you try make_mz7020_som_base.sh

As I mentioned, running make_mz7020_som_base.sh with all the same setup as you reproduces the issue of those missing lines.
Here's where it gets interesting though. I then checked out my branch wp-mz7020-fix-irq in project-spec/meta-avnet and ran rebuild_mz7020_som_base.shl to see if my changes would be picked up in pl.dtsi. Although the timestamp was updated, the changes were not reflected.

This tells me that pl.dtsi is not cleanly being regenerated.

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/Avnet/meta-avnet/pull/47#issuecomment-991316380>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCNSOABDBSNM4GKWPXRWETUQJYD3ANCNFSM5JULFRCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

wpatterson-witekio commented:
I just tried `petalinux-build -x mrproper -f` follwed by `rebuild_mz7020_som_base.sh` and  again the pl.dtsi was missing the interrupt-parent and interrupt-names, despite having them in the system-bsp.dtsi.  Have you tried `make_mz7020_som_base.sh` 

I would be very interested to see if you are seeing the same as me when building with the `make` script

narrucmot commented:
I did a fresh build of the underlying Vivado hw project:
$ cd /path/to/hdl/scripts
$ time vivado -mode batch -source ./make_mz7020_som_base.tcl

Followed by a fresh build of the PetaLinux BSP:
$ cd ../../petalinux/scripts
$ time ./make_mz7020_som_base.sh

Is your Vivado project stale?  If the Vivado project hasn’t been rebuilt since you last did a fresh git pull in the hdl repo then that could lead to the result you are seeing.  The PetaLinux build script just looks to see if the hw project is there and it imports the <>.xsa file it finds.  It has no idea if the <>.xsa file is valid/fresh or not.  The PetaLinux build script will only build the Vivado hw platform if it cannot be found.

--Tom


From: wpatterson-witekio ***@***.***>
Sent: Monday, December 13, 2021 4:09 PM
To: Avnet/meta-avnet ***@***.***>
Cc: Curran, Tom (EM) ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [Avnet/meta-avnet] Fixing device tree for irq chip initialization for mz7020 (PR #47)


I just tried petalinux-build -x mrproper -f follwed by rebuild_mz7020_som_base.sh and again the pl.dtsi was missing the interrupt-parent and interrupt-names, despite having them in the system-bsp.dtsi. Have you tried make_mz7020_som_base.sh

I would be very interested to see if you are seeing the same as me when building with the make script

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/Avnet/meta-avnet/pull/47#issuecomment-992911009>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCNSOHAB2YHLI37SCVUIFTUQZOFLANCNFSM5JULFRCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

wpatterson-witekio commented:
Rebuilding the vivado project fixed the issue, I'll close this PR

narrucmot commented:
Excellent!  Sorry you had so much trouble.

Time to build and test for release!

--Tom


From: wpatterson-witekio ***@***.***>
Sent: Tuesday, December 14, 2021 12:37 PM
To: Avnet/meta-avnet ***@***.***>
Cc: Curran, Tom (EM) ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [Avnet/meta-avnet] Fixing device tree for irq chip initialization for mz7020 (PR #47)


Rebuilding the vivado project fixed the issue, I'll close this PR

—
You are receiving this because your review was requested.
Reply to this email directly, view it on GitHub<https://github.com/Avnet/meta-avnet/pull/47#issuecomment-993818479>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ADCNSOCER4GMOVROFXYI4LLUQ56C5ANCNFSM5JULFRCQ>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
